composeApp/src/commonMain/kotlin/io/github/couchtracker/db/user/model/icon/DbIcon.kt

```diff
@@ -0,0 +1,52 @@
+package io.github.couchtracker.db.user.model.icon
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.BrokenImage
+import io.github.couchtracker.db.user.CouchTrackerUri
+import io.github.couchtracker.utils.Icon
+import io.github.couchtracker.utils.toIcon
+
+/**
+ * Represents an icon that can be stored in the database
+ */
+sealed interface DbIcon {
+
+    val icon: Icon
+
+    fun toCouchTrackerUri(): CouchTrackerUri
+
+    data class Default(val defaultIcon: DbDefaultIcon) : DbIcon {
+        override val icon get() = defaultIcon.icon
+        override fun toCouchTrackerUri() = CouchTrackerUri.Authority.ICON.uri("$DEFAULT_ICON_PATH/${defaultIcon.id}")
+    }
+
+    data class UnknownDefault(val originalUri: CouchTrackerUri) : DbIcon {
+        override val icon get() = Icons.Default.BrokenImage.toIcon()
+        override fun toCouchTrackerUri() = originalUri
+    }
+
+    companion object {
+        private const val DEFAULT_ICON_PATH = "default"
```

Comment:
Question: will there ever be something other than "default"? I think it's implied, given the `couch-tracker://`